### ISSUE TITLE
don't assign stdstar and sky next to each other on CCD

### ISSUE THREAD

**Issue by sbailey** [desihub/fiberassign#97]
We shouldn't assign standard stars and sky locations to fibers that are neighboring *on the CCD* due to the possibility of the bright standard star contaminating the neighboring sky due to unmodeled spectrograph PSF wings.  Currently fiber assignment only knows about the locations of positioners *on the focal plane*, but doesn't know where those fibers go in the spectrograph CCDs.  It is quite possible for fibers that are not neighbors on the focal plane to be neighbors on the CCD.

desimodel/data/focalplane/fiberpos.* contains the mapping (to be updated after the real focal plane is assembled) between the fiber number and the focal plane location.  Fiber assign should read this file and use that information to avoid assigning sky and standards next to each other.

To repeat: this is about not assigning sky and stds next to each other *on the CCD*, not about their relative locations on the focal plane.

**Comment by rncahn** [comment]
Hmm.  Not nice.

**Comment by sbailey** [comment]
This one is tricky, and would be the first case where fiberassign needs to know about where the fibers go on the CCD.  For starters, let's evaluate how often this happens by random chance.  If it is rare anyway, we could just exclude those sky fibers in the downstream analysis and not worry about it at the level of fiber assignment.

**Comment by rncahn** [comment]
8% of the fibers go to sky fibers.  The odds that a standard star fiber is
adjacent to a sky fiber is about 16%.

**Comment by forero** [comment]
That's right. So, for 100 star fibers + 400 sky fibers, we roughly have 16 (sky,star) pairs contiguous on the CCD. Is that too bad?

**Comment by tskisner** [comment]
Updating this for Spring 2020 code:  the fiberassign Hardware model does have access to the fiber mapping.  Implementing this avoidance logic is quite complicated, so we should really see if this is a problem in practice before working on it.
